data-protocols/ids/ids-transform-v1/src/main/java/org/eclipse/dataspaceconnector/ids/transform/type/asset/AssetFromIdsArtifactTransformer.java

```diff
@@ -45,22 +46,25 @@ public Class<Asset> getOutputType() {
             return null;
         }
 
-        IdsId idsId;
-        try {
-            idsId = IdsIdParser.parse(object.getId().toString());
-        } catch (IllegalArgumentException e) {
-            context.reportProblem(String.format("cannot read IdsId from artifact (id: %s)", object.getId().toString()));
-            return null;
+        var result = IdsId.from(object.getId());
+        if (result.failed()) {
+            context.reportProblem("id of incoming IDS artifact expected to be not null");
```

Comment:
If the result is failed, the content will be null, so there will be a NPE at line 55, better to return null here? Same for other transformers

Reply:
this id mess, sry, I will have a detailed look again. I think this is only a problem for the asset transformers

Reply:
@ndr-brt Should be fine now